### ISSUE TITLE
Autocomplete relative links and anchors in server mode

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#434]
### Problem or motivation

`rumdl` can tell you if a relative link is or isn't valid (rule MD057 (https://github.com/rvben/rumdl/blob/158b0e0cc930363b6fc186be2088c851a75e995e/docs/md057.md), but doesn't offer any help to find valid links.

Because `rumdl` can work as an LSP server (`rumdl server`), it might be able to offer more advanced LSP functionality.

### Proposed solution

Given `rumdl` has logic for determining if a relative link is valid for rule MD057, it seems as though `rumdl` could:

1. Suggest links as-you-type when it detects you're typing a link in a Markdown file. For example:
   ```markdown
   This is a sentence that [has a link](...
   ```
1. When you select one of the suggestions, auto-completes the URL. For example:
   ```markdown
   This is a sentence that [has a link](linked_file.md
   ```
1. If you keep typing with a `#`, begin to suggest section headings as anchors (perhaps relying on logic for rule MD051: https://github.com/rvben/rumdl/blob/158b0e0cc930363b6fc186be2088c851a75e995e/docs/md051.md). For example:
   ```markdown
   This is a sentence that [has a link](linked_file.md#section-heading
   ```

It would be good if it could figure out when to add the trailing `)` to close the link. Not sure what the heuristic would be for that. Folks could be left to do that manually, or perhaps there's a key combination that would do it.

This might only make sense for folks that are using inline links only, so could make use of configuration for rule MD054 (https://github.com/rvben/rumdl/blob/158b0e0cc930363b6fc186be2088c851a75e995e/docs/md054.md). Or perhaps it could work anywhere text that looks like any type of URL is typed.

This might functionality might default to off, so could be controlled by:

- A new setting in MD057 to turn suggestions on.
- Perhaps even in MD051 to also turn on anchor suggestions.

### Feature category

LSP / editor integration

**Comment by rvben** [comment]
Hi @eread, thanks for the suggestion! This is implemented since v0.1.26. The LSP server now provides:

- **File path completion** inside `[text](...)` — suggests relative paths to markdown files in the workspace
- **Heading anchor completion** after `#` — suggests section anchors from the target file (e.g., `[text](file.md#section-heading`)

Completion is triggered automatically by `(` and `#` characters. No configuration needed — it works out of the box with `rumdl server`.

Separately, we're looking at go-to-definition and find-references for links (#447).

**Comment by eread** [comment]
> Hi [@eread](https://github.com/eread), thanks for the suggestion! This is implemented since v0.1.26. The LSP server now provides:
> 
> * **File path completion** inside `[text](...)` — suggests relative paths to markdown files in the workspace
> * **Heading anchor completion** after `#` — suggests section anchors from the target file (e.g., `[text](file.md#section-heading`)
> 
> Completion is triggered automatically by `(` and `#` characters. No configuration needed — it works out of the box with `rumdl server`.
> 
> Separately, we're looking at go-to-definition and find-references for links ([#447](https://github.com/rvben/rumdl/issues/447)).

@rvben You're welcome. It works well!

**Comment by trepidacious** [comment]
@rvben Sorry to add this to a closed issue, but did there end up being a way to disable this? I get some very weird completions with initial `../` that I really don't want. I'm writing an LSP for a specific application of markdown where I want to provide my own completions based on context etc., but it's still great to use rumdl as well for markdown linting.

**Comment by rvben** [comment]
Yes, you can turn this off. rumdl's link path/anchor completions are controlled by the `enableLinkCompletions` LSP setting (default `true`); set it to `false` and rumdl stops offering them while keeping linting, formatting, and code actions. There's a separate `enableLinkNavigation` for hover/definition/references/rename on links, if you also want your own LSP to own those.

These are top-level LSP initialization options (camelCase, Ruff-style). In Neovim:

```lua
vim.lsp.config("rumdl", {
  cmd = { "rumdl", "server" },
  filetypes = { "markdown" },
  init_options = {
    enableLinkCompletions = false,
    enableLinkNavigation = false,
  },
})
```

Apologies these weren't documented - that's the real miss here. I've added an "LSP settings" section to the docs and tightened it so that with `enableLinkCompletions` off, rumdl no longer registers the link-target trigger characters (`(` `#` `/` `.` `-`) at all (code-block language completion still works). The setting itself already works in 0.2.24, so you can use it right now.

**Comment by trepidacious** [comment]
Thanks that's perfect, I've set that up following the [docs](https://rumdl.dev/lsp/#lsp-settings) and it now works great with my LSP.

I'm in Helix, so just for anyone finding this in future, the settings go in `languages.toml`, under the [`config` field](https://docs.helix-editor.com/languages.html#language-server-configuration). 

I used:

```toml
   [language-server.rumdl]
   command = "rumdl"
   args = ["server"]
   config = { enableLinkCompletions = false, enableLinkNavigation = false }
```

**Comment by rvben** [comment]
@trepidacious, thanks for the feedback and verification! 🚀
